notesapi/v1/tests/test_views.py

```diff
@@ -346,6 +357,64 @@ def test_create_multirange(self):
         del annotation['created']
         self.assertEqual(annotation, note)
 
+    @ddt.data(
+        (6, False),
+        (4, True)
+    )
+    @ddt.unpack
+    @patch('django.conf.settings.MAX_ANNOTATIONS_PER_COURSE', 5)
+    def test_create_maximum_allowed(self, num_notes, should_create):
+        """
+        Tests user can not create more than allowed notes/annotations per course
+        """
+        for i in xrange(num_notes - 1):
+            kwargs = {'text': 'Foo_{}'.format(i)}
+            self._create_annotation(**kwargs)
+
+        # Creating more notes should result in 400 error
+        kwargs = {'text': 'Foo_{}'.format(num_notes)}
+        response = self._create_annotation_without_assert(**kwargs)
+
+        if not should_create:
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST, 'Creating more than allowed notes')
+            self.assertEqual(
+                response.data['error_msg'],
+                u'You can create up to {0} notes.'
+                u' You must remove some notes before you can add new ones.'.format(settings.MAX_ANNOTATIONS_PER_COURSE)
+            )
+        else:
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    @patch('django.conf.settings.MAX_ANNOTATIONS_PER_COURSE', 5)
+    def test_create_maximum_other_user_course(self):
+        # if user tries to create notes in another course it should succeed
+        for i in xrange(5):
+            kwargs = {'text': 'Foo_{}'.format(i)}
+            self._create_annotation(**kwargs)
+
+        # Creating more notes should result in 400 error
+        kwargs = {'text': 'Foo_{}'.format(6)}
+        response = self._create_annotation_without_assert(**kwargs)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST, 'Creating more than allowed notes')
+        self.assertEqual(
+            response.data['error_msg'],
+            u'You can create up to {0} notes.'
+            u' You must remove some notes before you can add new ones.'.format(settings.MAX_ANNOTATIONS_PER_COURSE)
+        )
```

Comment:
L391-402 are duplicated in `test_create_maximum_allowed`, can we refactor both tests to avoid the duplication. Even i think we don't need `test_create_maximum_allowed` test.